bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveSwitchAllConverter.java

```diff
@@ -0,0 +1,119 @@
+ * {@link ZWaveSwitchAllCommandClass}. Implements All On/All Off.
+ * @author Pedro Paixao
+ * @since 1.8.0
+ */
+public class ZWaveSwitchAllConverter extends ZWaveCommandClassConverter<ZWaveSwitchAllCommandClass> {
+    private static final Logger logger = LoggerFactory.getLogger(ZWaveSwitchAllConverter.class);
+    private static final int REFRESH_INTERVAL = 0;
+
+    /**
+	 * Constructor. Creates a new instance of the {@link ZWaveSwitchAllConverter} class.
+	 * @param controller the {@link ZWaveController} to use for sending messages.
+	 * @param eventPublisher the {@link EventPublisher} to use to publish events.
+	 */
+    public ZWaveSwitchAllConverter(ZWaveController controller, EventPublisher eventPublisher) {
+        super(controller, eventPublisher);
+        this.addStateConverter(new BinaryDecimalTypeConverter());
+        this.addStateConverter(new BinaryPercentTypeConverter());
+        this.addStateConverter(new IntegerOnOffTypeConverter());
+        this.addStateConverter(new IntegerOpenClosedTypeConverter());
+        this.addStateConverter(new IntegerUpDownTypeConverter());
+        this.addCommandConverter(new BinaryOnOffCommandConverter());
+    }
+
+    /**
+	 * {@inheritDoc}
+	 */
+    @Override
+    public SerialMessage executeRefresh(ZWaveNode node, 
+    		ZWaveSwitchAllCommandClass commandClass, int endpointId, Map<String, String> arguments) {
+        logger.debug("NODE {}: Generating poll message for {}, endpoint {}", new Object[]{node.getNodeId(), commandClass.getCommandClass().getLabel(), endpointId});
+        return node.encapsulate(commandClass.getValueMessage(), commandClass, endpointId);
+    }
+
+    /**
+	 * {@inheritDoc}
+	 */
+    @Override
+    public void handleEvent(ZWaveCommandClassValueEvent event, Item item, Map<String, String> arguments) {
+    	ZWaveStateConverter<?,?> converter = this.getStateConverter(item, event.getValue());
+    	
+        if (converter == null) {
+            logger.warn("NODE {}: No converter found for item = {}, node = {} endpoint = {}, ignoring event.", new Object[]{event.getNodeId(), item.getName(), event.getEndpoint()});
+            return;
+        }
+        
+        State state = converter.convertFromValueToState(event.getValue());
+        this.getEventPublisher().postUpdate(item.getName(), state);
+    }
+
+    /**
+	 * {@inheritDoc}
+	 */
+    @Override
+    public void receiveCommand(Item item, Command command, ZWaveNode node, ZWaveSwitchAllCommandClass commandClass, int endpointId, Map<String, String> arguments) {
+    	ZWaveCommandConverter<?,?> converter = this.getCommandConverter(command.getClass());
+        
+    	if (converter == null) {
+            logger.warn("NODE {}: No converter found for item = {}, endpoint = {}, ignoring command.", new Object[]{node.getNodeId(), item.getName(), endpointId});
+            return;
+        }
+        
+    	Integer cmd = (Integer)converter.convertFromCommandToValue(item, command);
+        
+    	if (cmd == 0x00) {
```

Comment:
This always sends the command to the controller, which in turn always sends it to all nodes. While this might be the intention of the command class, in your example, you use 
```
Switch AllOnOff "All On/Off" {zwave="4:command=SWITCH_ALL"}
```

This configures node 4 - so I would have expected this command to be sent only to node 4? Wouldn't it make more sense to send to the node the user requested? If they want it sent to all nodes, then they can use the controller node id, and then you should add a check here to see if the node is to the controller (in which case do what you have here) or to another node, in which case you send just to that node?